CMakeLists.txt

```diff
@@ -80,6 +80,7 @@ if(NOT CMAKE_BUILD_TYPE)
         set(CMAKE_BUILD_TYPE "Release")
 endif()
 
+# FIXME: ASAN is currently broken
```

Comment:
If you can build successfully but it's the address sanitizer fails at runtime, please share the logs.

Reply:
See https://github.com/netdata/netdata/actions/runs/8249065227/job/22560690177, CI unit test logs from the commit before I disabled ASAN in the unit tests job. Disabling ASAN fixed things, and the error messages indicated some issue with linking order for ASAN.

I was able to reproduce the same issue locally both with the job as-is (prepping an Ubuntu 22.04 container with the required tooling and then running the script) and with ASAN enabled via CFLAGS instead of the CMake option (to confirm that it was not some quirk of how we were handling it in CMake).

Reply:
Regarding ASAN:

- Forcing use of a static ASAN runtime (default behavior with Clang, enabled with GCC by adding `-static-libasan` to the linking flags) resolves the runtime failures we were seeing, but results in different failures because protoc itself triggers ASAN warnings.
- Skipping ASAN for protoc itself results in linking errors, because it links against libprotobuf (and transitively against Abseil), which was built with ASAN.
- Trying to use shared libraries for protobuf/protoc instead of static ones results in different linking problems.

The only approach that I’ve found which works is to completely disable ASAN for Abseil and Protobuf. As such, I’ve updated the PR to just use that approach.